gui/builtinViews/fittingView.py

```diff
@@ -408,6 +408,9 @@ def swapItems(self, x, y, srcIdx):
             if mod1.slot != mod2.slot:
                 return
 
+            if not getattr(mod2, "modPosition"):
+                self.mods.modPosition = mod2.modPosition = dstRow
+
```

Comment:
> Long story short there are times when we can have modPosition be empty. This is just a check to see if it is

When will `modPosition` be empty? If we're in the `swapItems` methods, then that means we have two modules to swap and they will both have this property simply due to the fact that they are active modules. It's okay to assume we return something here, unless you noted it actually breaking during normal use. In which case, if they don't have it, I'd like for it to break so we know about it rather than trying to deduce which position the module is supposed to be in.

Furthermore, we shouldn't be setting `modPosition`. It doesn't have a setter and shouldn't have one - this is supposed to be a read-only property that returns the modules index in the fit's module list. I think this is being done all wrong.

Additionally, I'm curious on two things:
* Why `getattr(mod2, "modPosition")` instead of simply `mod2.modPosition`?
* Why are you setting `self.mods.modPosition`? `self.mods` is a list of modules...